app/main/views/suppliers.py

```diff
@@ -422,3 +377,103 @@ def register_interest_in_framework(supplier_id, framework_slug):
             return jsonify(message="Database Error: {0}".format(e)), 400
 
         return jsonify(frameworkInterest=interest_record.serialize()), 201
+
+
+@main.route('/suppliers/<supplier_id>/frameworks/<framework_slug>', methods=['PUT'])
+def register_framework_interest(supplier_id, framework_slug):
+    updater_json = validate_and_return_updater_request()
+
+    framework = Framework.query.filter(
+        Framework.slug == framework_slug
+    ).first_or_404()
+
+    supplier = Supplier.query.filter(
+        Supplier.supplier_id == supplier_id
+    ).first()
+
+    if supplier is None:
+        abort(404, "supplier_id '{}' not found".format(supplier_id))
+
+    interest_record = SupplierFramework.query.filter(
+        SupplierFramework.supplier_id == supplier_id,
+        SupplierFramework.framework_id == framework.id
+    ).first()
+
+    if interest_record:
+        return jsonify(frameworkInterest=interest_record.serialize()), 200
+
+    if framework.status != 'open':
+        abort(400, "'{}' framework is not open".format(framework_slug))
+
+    interest_record = SupplierFramework(
+        supplier_id=supplier_id,
+        framework_id=framework.id
+    )
+    audit_event = AuditEvent(
+        audit_type=AuditTypes.register_framework_interest,
+        user=updater_json.get('updated_by'),
+        data={'supplierId': supplier.supplier_id, 'frameworkSlug': framework_slug},
+        db_object=supplier
+    )
+
+    try:
+        db.session.add(interest_record)
+        db.session.add(audit_event)
+        db.session.commit()
+    except IntegrityError as e:
+        db.session.rollback()
+        return jsonify(message="Database Error: {0}".format(e)), 400
+
+    return jsonify(frameworkInterest=interest_record.serialize()), 201
+
+
+@main.route('/suppliers/<supplier_id>/frameworks/<framework_slug>', methods=['POST'])
+def update_supplier_framework_details(supplier_id, framework_slug):
+    updater_json = validate_and_return_updater_request()
+
+    framework = Framework.query.filter(
+        Framework.slug == framework_slug
+    ).first_or_404()
+
+    supplier = Supplier.query.filter(
+        Supplier.supplier_id == supplier_id
+    ).first()
+
+    if supplier is None:
+        abort(404, "supplier_id '{}' not found".format(supplier_id))
+
+    json_payload = get_json_from_request()
+    json_has_required_keys(json_payload, ["update"])
```

Comment:
I think we should stick with the pattern we use for service updates with update key matching the response key, so this should be `"frameworkInterest"`